source/Camera/InfoUtilOther/ZividBenchmark/ZividBenchmark.cpp

```diff
@@ -459,6 +472,82 @@ namespace
         printCapture2DResults(allDurations);
     }
 
+    Zivid::Frame assistedCapture(Zivid::Camera &camera)
+    {
+        const auto parameters = Zivid::CaptureAssistant::SuggestSettingsParameters{
+            Zivid::CaptureAssistant::SuggestSettingsParameters::AmbientLightFrequency::none,
+            Zivid::CaptureAssistant::SuggestSettingsParameters::MaxCaptureTime{ std::chrono::milliseconds{ 800 } }
+        };
+        const auto settings = Zivid::CaptureAssistant::suggestSettings(camera, parameters);
+        return camera.capture(settings);
+    }
+
+    template<typename DataType>
+    Duration copyDataTime(Zivid::Frame frame, DataType copyName)
+    {
+        std::vector<Duration> copyDataDurations;
+
+        auto pointCloud = frame.pointCloud();
+        const auto beforeCopyData = HighResClock::now();        
+        const auto tempCopy = pointCloud.copyData<decltype(copyName)>();
+        const auto afterCopyData = HighResClock::now();
+
+        return afterCopyData - beforeCopyData;
+    }
+
+    Duration copyDataTime(Zivid::Camera &camera, Zivid::Settings2D settings2D)
+    {
+        std::vector<Duration> copyDataDurations;
+
+        const auto beforeCopyData = HighResClock::now();
+        const auto frame2D = camera.capture(settings2D);
+        const auto image = frame2D.imageRGBA();
+        const auto afterCopyData = HighResClock::now();
+
+        return afterCopyData - beforeCopyData;
+    }
+
+    void benchmarkCopyData(Zivid::Camera &camera, const std::chrono::microseconds exposureTime, const size_t numFrames)
+    {
+        std::vector<Duration> copyDataDurations[9];
```

Comment:
make a variable called numData = 9 and use it everywhere in this function. It will be easier to add stuff.